tools/test.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Build and test the site content
+#
+# Requirement: html-proofer, jekyll
+#
+# Usage: See help information
+
+set -eu
+
+SITE_DIR="_site"
+
+_config="_config.yml"
+
+help() {
+  echo "Build, test and then deploy the site content to 'origin/<pages_branch>'"
+  echo
+  echo "Usage:"
+  echo
+  echo "   bash ./tools/deploy.sh [options]"
+  echo
+  echo "Options:"
+  echo '     -c, --config   "<config_a[,config_b[...]]>"    Specify config file(s)'
+  echo "     -h, --help               Print this information."
+}
+
+build() {
+  # clean up
+  if [[ -d $SITE_DIR ]]; then
+    rm -rf "$SITE_DIR"
+  fi
+
+  # build
+  JEKYLL_ENV=production bundle exec jekyll \
+    build \
+    --destination "$SITE_DIR" \
+    --config "$_config"
+}
+
+test() {
+  bundle exec htmlproofer \
+    --disable-external \
+    --allow_hash_href \
+    "$SITE_DIR"
+}
+
+main() {
+  build
+  test
+}
```

Comment:
Compare to the former `deploy.sh` script. This essentially covers all the same methods as the `--dry-run` option from before, so everything called after that point has just been removed.

Further, since this isn't in the business of actually backing up and deploying anything, I found it safe to go with the much simpler `build` and `test` options, without mucking about in extra SITE_DIR locations or renames.

That said, this script is now strictly a local tool. The Actions workflow runs the tests on its own, without invoking this script.